src/ptf/mask.py

```diff
@@ -22,13 +23,24 @@ def set_do_not_care(self, offset, bitwidth):
             self.mask[offsetB] = self.mask[offsetB] & (~(1 << (7 - offsetb)))
 
     def set_do_not_care_packet(self, hdr_type, field_name):
+        def get_raw_packet():
+            if PY2:
+                if hasattr(self.exp_pkt[hdr_type], "__bytes__"):
+                    return self.exp_pkt[hdr_type].__bytes__()
+            return bytes(self.exp_pkt[hdr_type])
```

Comment:
I think that you can simplify it to just `return bytes(self.exp_pkt[hdr_type])` as `bytes()` calls `__bytes__` anyway (tested it on both PY2 and PY3).

Reply:
Ok, corrected. I base my change on the usage of Scapies' ` raw` method. https://github.com/secdev/scapy/blob/master/scapy/compat.py#L270